lib/libzpool/uzfs_io.c

```diff
@@ -82,6 +84,10 @@ uzfs_write_data(zvol_state_t *zv, char *buf, uint64_t offset, uint64_t len,
 	uint64_t orig_offset = offset;
 	char *mdata = NULL, *tmdata = NULL, *tmdataend = NULL;
 
+	if (uzfs_write_size) {
+		// align it in the multiple of blocksize
+		blocksize = (uzfs_write_size / blocksize) * blocksize;
```

Comment:
ceiling would be required here so that blocksize won't be 0 if uzfs_write_size happens to be less than blocksize